test/framework/containers.py

```diff
@@ -244,7 +246,18 @@ def test_end2end_singularity_recipe_config(self):
             regex = re.compile('^' + pattern, re.M)
             self.assertFalse(regex.search(txt), "Pattern '%s' should not be found in: %s" % (regex.pattern, txt))
 
-        regex = re.compile("^id easybuild", re.M)
+        for pattern in ["id easybuild", eb_pattern]:
+            regex = re.compile('^' + pattern, re.M)
+            self.assertTrue(regex.search(txt), "Pattern '%s' should be found in: %s" % (regex.pattern, txt))
+
+        remove_file(test_container_recipe)
+
+        # options can be passed to 'eb' command in recipe via 'eb_args' keyword
+        args[-1] = 'bootstrap=yum,osversion=7.6.1810,eb_args=--debug -l'
```

Comment:
Looks strange not to have
```
args[-1] = 'bootstrap=yum,osversion=7.6.1810,eb_args="--debug -l"'
```

Reply:
Or does that have an impact in the regex later?

Reply:
It looks strange indeed, but that's mainly because there's no shell instance that is interpreting this (so the `"` quotes wouldn't get stripped out).

Outside of the tests, you would indeed have to use quotes to preserve the space in the value for `eb_args`:

```
eb --container-config bootstrap=yum,osversion=7.6.1810,eb_args="--debug -l" ...
```

or

```
eb --container-config "bootstrap=yum,osversion=7.6.1810,eb_args=--debug -l" ...
```

The latter explains it better: we're basically only passing the value within the `"..."` to `args[-1]` in the test.